internal/execute/watcher.go

```diff
@@ -54,7 +54,7 @@ func (w *watcher) hasErrorsInTsConfig() bool {
 			}
 			return true
 		}
-		if !reflect.DeepEqual(w.options.CompilerOptions(), configParseResult.CompilerOptions()) {
```

Comment:
This is definitely not correct, sorry. We need to be comparing all of the options, as using only this subset means we are missing many more options that matter for loading things or any checking option.

I entirely forgot that this code was here. I'll come up with a better fix.